server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotResponse.java

```diff
@@ -93,4 +100,58 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endObject();
         return builder;
     }
+
+    public static CreateSnapshotResponse fromXContent(XContentParser parser) throws IOException {
+        CreateSnapshotResponse createSnapshotResponse = new CreateSnapshotResponse();
+
+        parser.nextToken(); // move to '{'
+
+        if (parser.currentToken() != Token.START_OBJECT) {
+            throw new IllegalArgumentException("unexpected token [" + parser.currentToken() + "], expected ['{']");
+        }
+
+        parser.nextToken(); // move to 'snapshot' || 'accepted'
+
+        if ("snapshot".equals(parser.currentName())) {
+            createSnapshotResponse.snapshotInfo = SnapshotInfo.fromXContent(parser);
+        } else if ("accepted".equals(parser.currentName())) {
+            parser.nextToken(); // move to 'accepted' field value
+
+            if (parser.booleanValue()) {
+                // ensure accepted is a boolean value
+            }
+
+            parser.nextToken(); // move past 'true'/'false'
+        } else {
+            throw new IllegalArgumentException("unexpected token [" + parser.currentToken() + "] expected ['snapshot', 'accepted']");
```

Comment:
parsing here is very strict. while strictness is generally a good thing, it hinders forward compatibility in this specific case when parsing back responses in the client. That's why parsing should also be tested with `supportsUnknownFields` set to `true` unless there are good reasons not to do so.
What if one day we add a new field to the response, which is a perfectly backwards compatible change? The client would break with such a response, while it should ignore the additional fields till the newer client is used, which will be able to read such new field.

Reply:
Switched to a lenient ObjectParser.